### ISSUE TITLE
Wide review tracker

### ISSUE THREAD

**Issue by anssiko** [webmachinelearning/webnn#239]
### About

This is a meta issue to track [wide review](https://www.w3.org/2021/Process-20211102/#wide-review) for the [Web Neural Network API](https://www.w3.org/TR/webnn/).

An important part of wide review is horizontal review from W3C's [key horizontal groups](https://www.w3.org/Guide/documentreview/#how_to_get_horizontal_review) listed below. Also feedback from other stakeholders is equally important and we're documenting some of such signals in this issue too. Additional pointers are welcome via comments.

>Legend:
>🔴 Review request not submitted
>🟡 Review request submitted
>🔵 Review feedback received
>🟢 Review closed as completed

### Horizontal groups

🟢 ♿ **Accessibility**

- [x] Work through <a rel="nofollow" class="external text" href="https://w3c.github.io/apa/fast/checklist.html">this questionnaire</a> - https://github.com/webmachinelearning/webnn/issues/261
    - [x] <a rel="nofollow" class="external text" href="https://github.com/w3c/a11y-request/issues/new/choose">request a review via GitHub</a> from <a rel="nofollow" class="external text" href="https://www.w3.org/WAI/APA/">APA</a> - https://github.com/w3c/a11y-request/issues/26
    - [x] Review confirmed completed https://github.com/w3c/a11y-request/issues/26#issuecomment-1260986777 

🟢 📐 **Architecture**

- [x] Ask the <a rel="nofollow" class="external text" href="https://www.w3.org/2001/tag/">TAG</a> for review; see <a rel="nofollow" class="external text" href="https://tag.w3.org/workmode/">how to work with the TAG</a>
    - [x] full review https://github.com/w3ctag/design-reviews/issues/570 completed
    - [x] delta review https://github.com/w3ctag/design-reviews/issues/771 (Oct'21-Sep'22)
    - [x] delta review 2 https://github.com/w3ctag/design-reviews/issues/933 (Mar'23-Mar'24) - NB: any additional off-cycle feedback is also considered by the group as per our commitment for close coordination with horizontal groups
    - [ ] If you are developing javascript APIs you may also want to ask <a rel="nofollow" class="external text" href="mailto:public-script-coord@w3.org">public-script-coord@w3.org</a>, a technical discussion list shared by W3C and ECMA's TC 39

🟢 🌍 **Internationalisation**

- [x] Read the <a rel="nofollow" class="external text" href="https://www.w3.org/International/review-request">Request a review</a> page
- [x] work through the <a rel="nofollow" class="external text" href="https://w3c.github.io/i18n-drafts/techniques/shortchecklist">Short Checklist</a> https://github.com/webmachinelearning/webnn/issues/262
- [x] <a rel="nofollow" class="external text" href="https://github.com/w3c/i18n-request/issues/new/choose">request a review via GitHub</a> - https://github.com/w3c/i18n-request/issues/179 https://github.com/w3c/i18n-request/issues/223

🟢 🔍 **Privacy**

- [x] Write a "[Privacy Considerations](https://www.w3.org/TR/webnn/#privacy)" section for your document, taking into account the [Self-Review Questionnaire: Security and Privacy](https://w3ctag.github.io/security-questionnaire/), [Mitigating Browser Fingerprinting in Web Specifications](https://w3c.github.io/fingerprinting-guidance/), and [RFC6973](https://tools.ietf.org/html/rfc6973)
   - [x] Completed https://github.com/webmachinelearning/webnn/issues/119
   - [x] https://github.com/webmachinelearning/webnn/pull/271 
   - [x] Consider [privacy-tracker](https://github.com/webmachinelearning/webnn/issues?q=label%3Aprivacy-tracker+) issues
- [x] <a rel="nofollow" class="external text" href="https://github.com/w3cping/privacy-request/issues/new/choose">request a review via GitHub</a> from the <a rel="nofollow" class="external text" href="https://www.w3.org/Privacy/">Privacy Interest Group</a>
    - [x] Early Privacy Review Request [submitted](https://lists.w3.org/Archives/Public/public-privacy/2020OctDec/0038.html) and response [received](https://github.com/webmachinelearning/webnn/issues/119#issuecomment-772565085) with comments addressed
    - [x] Submit a new Review Request for any privacy-impacting changes since the early request, if any https://github.com/w3cping/privacy-request/issues/96
    - [x] PING feedback integrated https://github.com/webmachinelearning/webnn/issues/280
    - [x] Review confirmed completed https://github.com/w3cping/privacy-request/issues/96#issuecomment-1234419529

🟢 🔒 **Security**

- [x] Write a "[Security Considerations](https://www.w3.org/TR/webnn/#security)" section for your document, taking into account the <a rel="nofollow" class="external text" href="https://w3ctag.github.io/security-questionnaire/">Self-Review Questionnaire: Security and Privacy</a>
    - [ ] Consider [security-tracker](https://github.com/webmachinelearning/webnn/labels/security-tracker) issues 
- [x] <a rel="nofollow" class="external text" href="https://github.com/w3c/security-request/issues/new/choose">request a review via GitHub</a> https://github.com/w3c/security-request/issues/22
- [x] Review closed as completed https://github.com/w3c/security-request/issues/22#issuecomment-1209469579

### Other stakeholders

From [who to ask for review](https://www.w3.org/Guide/documentreview/#who_to_ask_for_review):
>Horizontal reviews [...] are only a subset of a full wide review, which must also include other stakeholders including Web developers, technology providers and implementers not active in the Working Group, external groups and standards organizations working on related areas, etc.

🟢 WebGPU Working Group https://github.com/gpuweb/gpuweb/issues/2500
  - [x] Positive feedback received at the [GPU Web F2F meeting 2023-02-16 (Member-only)](https://lists.w3.org/Archives/Member/internal-gpu/2023Feb/0009.html), see also [WebML WG 2023-03-02 meeting minutes](https://www.w3.org/2023/03/02-webmachinelearning-minutes.html#t02)

🟢 Web developers: [TensorFlow.js support](https://www.youtube.com/watch?v=GbgjafMdAIs&t=867s), [Microsoft DevRel positive signals](https://twitter.com/manekinekko/status/1466328676044263424?s=20)

🟢 Broader community: [workshop recommendation](https://www.w3.org/2020/06/machine-learning-workshop/report.html)

🟢 Implementers: [positive signals](https://www.w3.org/blog/2021/04/w3c-launches-the-web-machine-learning-working-group/), [Chromium Intent to Prototype](https://groups.google.com/a/chromium.org/g/blink-dev/c/PD6TDMDS9mg/m/N3MrigMyCAAJ)

**Comment by anssiko** [comment]
Per our discussion in https://www.w3.org/2024/01/25-webmachinelearning-minutes.html#t01 I have initiated Architecture/TAG and Privacy delta wide reviews highlighting relevant new information since their previous review and informed also other horizontal groups (a11y, i18n, Security) of our plan to publish a new CR Snapshot during Q1'24.

I piggypacked on the existing review issues, links in the initial comment. Thanks to the WG for inputs reflected in these requests. I will bring concrete feedback from these reviews to the attention of the WG for deliberation.

**Comment by anssiko** [comment]
The latest TAG delta review request refiled as a new issue to hopefully bump its priority in triage https://github.com/w3ctag/design-reviews/issues/933 over the earlier review request we initially appended (thanks @dontcallmedom!).

**Comment by anssiko** [comment]
I have informed the TAG of our recent CR Snapshot publication and that any additional off-cycle feedback is welcome and will be considered by the group in the most recent version of the spec. With that, I'll close this issue. We'll re-open this tracker when we do another coordinated wide review for a future version as appropriate.

On behalf of the entire group, I want to thank the horizontal groups for their review, contributions and partnership.

**Comment by anssiko** [comment]
I'm reopening this wide review tracker for a new round of review requests that are in flight now:

Legend:
- 🟡 Review request submitted
- 🔵 Review feedback received
- 🟢 Review closed as completed
---

🟢 ♿ Accessibility https://github.com/w3c/a11y-request/issues/105
  - Related discussion https://github.com/w3c/apa/issues/350
  - Status: PR approved by APA WG and merged: https://github.com/webmachinelearning/webnn/pull/869

🟢 📐 Architecture https://github.com/w3ctag/design-reviews/issues/1072
  - Review feedback https://github.com/w3ctag/design-reviews/issues/1072#issuecomment-2941185185
  - Related issues #821 #861
  - Status: completed with non-blocking related issues

🟢 🌍 Internationalisation https://github.com/w3c/i18n-request/issues/258
   - Review feedback https://github.com/webmachinelearning/webnn/issues/837 
   - Response https://github.com/webmachinelearning/webnn/pull/841 and https://github.com/webmachinelearning/webnn/pull/851
   - Status: completed

🟢 🔍 Privacy https://github.com/w3cping/privacy-request/issues/156
   - Comments https://github.com/w3cping/privacy-request/issues/156#issuecomment-2813166867
   - On the Privacy WG agenda https://github.com/w3c/privacywg/issues/11
   - Status: merged https://github.com/webmachinelearning/webnn/issues/875 https://github.com/webmachinelearning/webnn/issues/886
 
🟢 🔒 Security https://github.com/w3c/security-request/issues/85
   - Status: completed with no concerns https://github.com/w3c/security-request/issues/85#issuecomment-3347488569

It may take a while for the horizontal groups to pick up the reviews. To that end, fine-tuning of the review requests is possible and welcome. We've asked review feedback to be delivered within the next three months. Meanwhile, the work on the API spec continues as usual.

Edit: Updated with the latest.

**Comment by anssiko** [comment]
We passed the scrutiny and shipped the WebNN CR Snapshot on 22 January 2026. I'm closing this tracker and will re-open as appropriate for the next cycle.
